bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/StackRenderer.java

```diff
@@ -209,6 +216,13 @@ private void addItemToSet(List<CTabItem> itemsToSet, MPart parentParent) {
 		}
 	}
 
+	@Inject
+	@Optional
+	void subscribePerspectiveChanged(
+			@UIEventTopic(UIEvents.UILifeCycle.PERSPECTIVE_OPENED) org.osgi.service.event.Event event) {
```

Comment:
I tried this one. It works only if the perspective is opened via "Open perspective dialog". It does not work if I just click a perspective button (e.g. java or debug) in the toolbar. The even is not raised then.
There are other events in UIEvents.Perspective (e.g. TOPIC_ALL). But is is also not raised.

Basically I would need a notification on startup and on perspective change. But without an osgi dependency to another plugin.

Any ideas?

Reply:
Ok, I have created a new ui event called PERSPECTIVE_SWITCHED.
With that it works.